docs(tcp): add documentation for k6/x/tcp extension and its API methods (#2233)
grafana/k6-docs: docs/sources/k6/next/javascript-api/k6-x-tcp/socket/destroy.md

---
title: 'Socket.destroy()'
description: 'Close and destroy the TCP socket'
weight: 41
---

# Socket.destroy()

> Closes the connection and destroys the socket. Once destroyed, the socket transitions to the `'destroyed'` state and cannot be reused. A `close` event is emitted after the socket is destroyed.

Reviewer: ```suggestion
Closes the connection and destroys the socket. After you call `destroy()`, the socket transitions to the `'destroyed'` state, and you can't reuse it. A `close` event is emitted after the socket is destroyed.
```
